reactor-core/src/test/java/reactor/core/publisher/AbstractOnDiscardShouldNotLeakTest.java

```diff
@@ -40,7 +42,7 @@ public AbstractOnDiscardShouldNotLeakTest(boolean conditional, boolean fused) {
         this.fused = fused;
     }
 
-    abstract Flux<Tracked<?>> transform(Flux<Tracked<?>> upstream);
+    protected abstract Publisher<Tracked<?>> transform(TestPublisher<Tracked<?>>... upstreams);
```

Comment:
if possible (depends on whether or not future test cases will absolutely need the whole array of publishers in a single array variable), I would suggest the following signature:

```java
protected abstract Publisher<Tracked<?>> transform(TestPublisher<Tracked<?>> firstUpstream, TestPublisher<Tracked<?>>... otherUpstreams);
```